config/package-is-installable/src/getSystemNodeVersion.ts

```diff
@@ -6,7 +6,7 @@ export function getSystemNodeVersionNonCached () {
   if (process['pkg'] != null) {
     return execa.sync('node', ['--version']).stdout.toString()
   }
-  return process.version
+  return process.version.replace(/-.*$/, '')
```

Comment:
why is this change required? Is it because it currently fails when checking if the node.js version satisfies the range in the engines field? I think in that case this change should be done before passing the version to semver. This function may be used also to print the node.js version and in that case I think we should return the correct version. Also, the way you did it, doesn't cover line 7.

Reply:
Yes, I installed version `v21.0.0-nightly20230429c968361829` to test a function of node. When I run the project, its `engine.node` field value `^14.18.0 || >=16.0.0` check will fail. Running it with npm will work fine.
> I think in that case this change should be done before passing the version to semver. 

You mean maybe we should change here?
https://github.com/pnpm/pnpm/blob/5e0165c598c518cddc28be05385133afaa06b203/config/package-is-installable/src/checkEngine.ts#L24
>This function may be used also to print the node.js version and in that case I think we should return the correct version. Also, the way you did it, doesn't cover line 7.

I was negligent, I didn't think about these before. If we modify it in the above way, there is no need to change it here.

Reply:
Yes, that looks like the right place for the change.

Reply:
Ok, let me modify it.